src/BikeSharing.DomainLogic/BikeWorkout.cs

```diff
@@ -0,0 +1,258 @@
+"#weights",
+"#Getfit",
+"#weightloss",
+"#wod",
+"#aesthetic",
+"#squad",
+"#shreadding",
+"#personaltrainer",
+"#dreambitviral",
+"#quote",
+"#quotes",
+"#inspiring",
+"#motivation",
+"#fitnessquote",
+"#youcandoit",
+"#justbringit",
+"#dreambig",
+"#success",
+"#staypositive",
+"#noexcuses"
+    };
+
+        public string Username { get; set; }
+        public Gender Gender { get; set; }
+        public int Age { get; set; }
+        public double Weight { get; set; }
+        public double Height { get; set; }
+        public double BasalMetabolicRate { get; }
+        public List<Workout> Workouts { get; set; }
+
+        public Athlete(string username, Gender gender, int age, double weight, double height)
+        {
+            Username = username;
+            Gender = gender;
+            Age = age;
+            Weight = weight;
+            Height = height;
+            BasalMetabolicRate = GetBasalMetabolicRate();
+        }
+
+        public Athlete(string username, Gender gender, int age, double weight, double height, List<Workout> workouts)
+        {
+            Username = username;
+            Gender = gender;
+            Age = age;
+            Weight = weight;
+            Height = height;
+            BasalMetabolicRate = GetBasalMetabolicRate();
+            Workouts = workouts;
+        }
+
+        public void AddWorkout(params Workout[] w)
+        {
+            if (Workouts != null)
+            {
+                Workouts.AddRange(w);
+            }
+            else
+            {
+                Workouts = w.ToList();
+            }
+        }
+
+        private double GetBasalMetabolicRate()
+        {
+            switch (Gender)
+            {
+                case Gender.Male:
+                    return 66.47 + (13.75 * Weight * 0.453592) + (5.003 * Height * 2.54) - (6.755 * Age);
+                case Gender.Female:
+                    return 655.1 + (9.563 * Weight * 0.453592) + (1.850 * Height * 2.54) - (4.676 * Age);
+                default:
+                    return 0.0;
+            }
+        }
+
+        public double GetCaloriesBurned(Workout workout)
+        {
+            //var a = (Gender)5;
+            switch (Gender)
+            {
+                case Gender.Male:
+                    return (-55.0969 + (0.6309 * workout.AverageHeartRate) 
+                           + (0.1988 * Weight) + (0.2017 * Age) / 4.184) 
+                           * 60 * workout.Duration.TotalHours;
+                case Gender.Female:
+                    return (-20.4022 + (0.4472 * workout.AverageHeartRate) 
+                           - (0.1263 * Weight) + (0.074 * Age) / 4.184) 
+                           * 60 * workout.Duration.TotalHours;
+                default:
+                    return 0.0;
+            }
+        }
+
+        public (Workout workout, double calories) GetWeeksBestWorkout()
+        {
+            var lastWeekWorkouts = Workouts.Where(w => w.Date > DateTime.Now.Date.AddDays(-7));
```

Comment:
OR 
```C#
DateTime.Now.Date.Subtract(TimeSpan.FromDays(7))
```

---
In reply to: [102630093](https://github.com/kuhlenh/Productivity/pull/1#discussion_r102630093) [](ancestors = 102630093)